lib/application.js

```diff
@@ -505,21 +505,21 @@ function dataSourcesFromConfig(name, config, connectorRegistry, registry) {
     'can not create data source without config object');
 
   if (typeof config.connector === 'string') {
-    name = config.connector;
-    if (connectorRegistry[name]) {
-      config.connector = connectorRegistry[name];
+    var connectorName = config.connector;
+    if (connectorRegistry[connectorName]) {
+      config.connector = connectorRegistry[connectorName];
     } else {
-      connectorPath = path.join(__dirname, 'connectors', name + '.js');
+      connectorPath = path.join(__dirname, 'connectors', connectorName + '.js');
 
       if (fs.existsSync(connectorPath)) {
         config.connector = require(connectorPath);
       }
     }
     if (config.connector && typeof config.connector === 'object' && !config.connector.name)
-      config.connector.name = name;
+      config.connector.name = connectorName;
   }
 
-  return registry.createDataSource(config);
+  return registry.createDataSource(name, config);
```

Comment:
@nitro404 Thank you for the contribution.

```js
"myds": {
  "name": "temp",  // datasource name
  "connecor": "mysql",  // connector name
  "url": "localhost:3306/mydb"
}
```
Given the example above, I agree that `server.datasources.myds.name` implies the datasource name NOT the connector name.

Most of your change looks good to me 👍 , my only concern is Line 522, I think the first param here should be the connector name not datasource name:
```js
// IIUC, here
// `name` is datasource name
// `connectorName` is the connector name
return registry.createDataSource(connectorName, config);
```

It eventually instantiates `loopback-datasource-juggler/lib/datasource.js: DataSource(name, settings, modelConfig)`, my understanding is that function expects `name` to be a connector name.

Or you can omit it like the old code.
Does test fail if you keep it as `return registry.createDataSource( config);`?

Reply:
The jsdoc comment for `Registry.prototype.createDataSource()` says that the first argument `name` is an "Optional name" (not very useful) and then expects `options.connector` to hold the connector name:

https://github.com/strongloop/loopback/blob/0a4940e31ef99ffbae98f91812f631a7d9a524cc/lib/registry.js#L349-L359

Down the line, both `name` and `options` are passed to DataSource constructor. AFAICT, the `name` is treated as the datasource name (not a connector name) in juggler, with the exception of this part, which should not be triggered when a DataSource is created from LoopBack (at least I think so):

https://github.com/strongloop/loopback-datasource-juggler/blob/00cf01f901e5ff20f3a576f10f41779e70cf536f/lib/datasource.js#L371-L376

If the connector is not provided, then it seems like the name is treated as a connector name, but again, I believe loopback's Registry always converts string connector name to a function/object before calling DataSource constructor.

https://github.com/strongloop/loopback-datasource-juggler/blob/00cf01f901e5ff20f3a576f10f41779e70cf536f/lib/datasource.js#L378-L393